lib/timeline.js

```diff
@@ -84,6 +84,33 @@ Timeline.prototype.sendSharedPin = function (topics, pin, callback) {
 };
 
 /**
+ * Delete a pin from user's timeline
+ *
+ * @param {String} userToken
+ * @param {Pin} pin
+ * @return {callback} callback
+ */
+Timeline.prototype.deleteUserPin = function (userToken, pin, callback) {
+  if (typeof userToken !== 'string') {
+    return callback(new Error('Expected userToken to be a string.'));
```

Comment:
we should fix every method to not callback() until the next tick to avoid hard to debug issues related to callback timing. basically, async contracts should always be async. example:

Today's incorrect behavior:
```js
timeline.deleteUserPin(token, pin, function() {
  console.log(2);
});
console.log(1);
```
Under non-error circumstances, the code will output:

```
1
2
```

But, when invalid input is passed, the code will output:

```
2
1
```

which is very bad b/c it changes the order in which the users code will execute, thereby causing race conditions in the users production code <=== our fault :(